### ISSUE TITLE
No valid hash for installer/archive found, exiting...

### ISSUE THREAD

**Issue by dannykorpan** [mkorthof/chrupd#22]
Hi,

thank you for your script and work!
At the moment there is a problem with hash for installation the Ungoogled release from Marmaduke.

Command line
`chrupd.cmd -editor Ungoogled -channel stable -arch 64bit`

chrupd.log
```
2022-10-18 16:46:45 Start (pid:4200 name:powershell scheduler:0 v:20211002 )
2022-10-18 16:46:45 Could not find Chromium, initial installation will be done by the downloaded installer...
2022-10-18 16:46:45 Feed "chromium.woolyss.com" editor "Hibbiki" architecture "64-bit" channel "stable" 
2022-10-18 16:46:46 New Chromium version "106.0.5249.119" from 2022-10-12 is available (6 days ago)
2022-10-18 16:46:46 Downloading "https://github.com/Hibbiki/chromium-win64/releases/download/v106.0.5249.119-r1036826/mini_installer.sync.exe"
2022-10-18 16:46:46 Saving as: "C:\Users\DANNYK~1\AppData\Local\Temp\mini_installer.sync.exe"
2022-10-18 16:54:52 Start (pid:7936 name:powershell scheduler:0 v:20211002 )
2022-10-18 16:54:52 Could not find Chromium, initial installation will be done by the downloaded installer...
2022-10-18 16:54:52 Feed "chromium.woolyss.com" title "Ungoogled" editor "Marmaduke" architecture "64-bit" channel "stable" 
2022-10-18 16:54:52 No valid hash for installer/archive found, exiting...
2022-10-18 16:55:02 Start (pid:11784 name:powershell scheduler:0 v:20211002 )
2022-10-18 16:55:02 Could not find Chromium, initial installation will be done by the downloaded installer...
2022-10-18 16:55:02 Feed "chromium.woolyss.com" title "Ungoogled" editor "Marmaduke" architecture "64-bit" channel "stable" 
2022-10-18 16:55:03 No valid hash for installer/archive found, exiting...
2022-10-18 16:57:43 Start (pid:14044 name:powershell scheduler:0 v:20211002 )
2022-10-18 16:57:43 Currently installed Chromium version: "106.0.5249.119"
2022-10-18 16:57:43 Feed "chromium.woolyss.com" title "Ungoogled" editor "Marmaduke" architecture "64-bit" channel "stable" 
2022-10-18 16:57:43 No valid hash for installer/archive found, exiting...
```

Kind Regards

**Comment by mkorthof** [comment]
Hi,
Thanks for your kind words.
Indeed it does seem the hashes for latest Ungoogled release are missing from Woolyss RSS feed.
You could add `-ignHash` option to skip hash check.

**Comment by funlockz** [comment]
It seems that the sha1 was added to the latest release on RSS https://chromium.woolyss.com/feed/windows-64-bit but still the script fails when I try:
`chrupd -editor Marmaduke -arch 64bit -channel stable`
or 
`chrupd -editor Ungoogled -arch 64bit -channel stable`
could it be that they changed the rss format so the parsing of sha1 doesn't work anymore?

chrupd.log
```
2022-11-01 15:55:02 Start (pid:1332 name:powershell scheduler:0)
2022-11-01 15:55:02 Currently installed Chromium version: "105.0.5195.127"
2022-11-01 15:55:02 Checking: "chromium.woolyss.com", Editor: "Ungoogled", Architecture: "64-bit", Channel: "stable"
2022-11-01 15:55:03 No matching Chromium versions found
```

**Comment by nikkehtine** [comment]
> Hi, Thanks for your kind words. Indeed it does seem the hashes for latest Ungoogled release are missing from Woolyss RSS feed. You could add `-ignHash` option to skip hash check.

I get this error:
```powershell
.\chrupd.cmd -editor Marmaduke -arch 64bit -channel stable -ignHash
 Simple Chromium Updater ( chrupd.cmd 20211002 ) 

OS Detected: "Windows 10" (10.0, Workstation)
Task Scheduler Mode: 1 "Normal"
Logging to: "E:\Documents\code\git\chrupd\chrupd.log"
WARNING: Could not find Chromium, initial installation will be done by the downloaded installer...
Using the following settings:
Feed "chromium.woolyss.com" editor "Marmaduke" architecture "64-bit" channel "stable" 


(!) Ignoring hash. Could not verify checksum of installer/archive.
    Press any key to abort or "c" to continue...

    Waiting 30 seconds before continuing, 24s left 
New Chromium version "107.0.5304.88" from 2022-10-28 is available (4 days ago)
Downloading "https://github.com/macchrome/winchrome/releases/download/v107.5304.88-M107.0.5304.88-r1047731-Win64/107.0.5304.88_ungoogled_mini_installer.exe"
Saving as: "C:\Users\noire\AppData\Local\Temp\mini_installer.exe"
Get-FileHash : The term 'Get-FileHash' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the na
me, or if a path was included, verify that the path is correct and try again.
At line:1594 char:14
+ $fileHash = (Get-FileHash -Algorithm $cdataObj.hashAlgo "$saveAsPath" ...
+              ~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Get-FileHash:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
Ignoring hash, using hash from downloaded installer: ""
ERROR: Hash Algorithm is missing, exiting...
```

When I run the command itself:

```powershell
Get-FileHash C:\Users\noire\AppData\Local\Temp\mini_installer.exe

Algorithm       Hash                                                                   Path
---------       ----                                                                   ----
SHA256          722857FF0397540E19D33656A5C2DAF7DC75D7BBAD2EBD979C809D997E153823       C:\Users\noire\AppData\Local\Temp\mini_installer.exe
```

**Comment by mkorthof** [comment]
Should be fixed in latest version (20221105)
